packages/edit-site/src/store/actions.js

```diff
@@ -228,3 +230,83 @@ export function updateSettings( settings ) {
 		settings,
 	};
 }
+
+/**
+ * Reverts a template to its original theme-provided file.
+ *
+ * This works with the assumption that the template being reverted
+ * belongs to the current theme.
+ *
+ * @param {Object} template The template to revert.
+ */
+export function* revertTemplate( template ) {
+	const postType = yield controls.resolveSelect(
+		'core',
+		'getPostType',
+		'wp_template'
+	);
+
+	const currentTheme = yield controls.resolveSelect(
+		'core',
+		'getCurrentTheme'
+	);
+
+	if ( ! isTemplateRevertable( template, currentTheme?.stylesheet ) ) {
+		yield controls.dispatch(
+			'core/notices',
+			'createErrorNotice',
+			__( 'This template is not revertable' ),
+			{ type: 'snackbar' }
+		);
+		return;
+	}
+
+	try {
+		yield apiFetch( {
+			path: `/wp/v2/${ postType.rest_base }/${ template.id }`,
+			method: 'DELETE',
+		} );
```

Comment:
```suggestion
		yield controls.dispatch(
			'core',
			'deleteEntityRecord',
			'postType',
			'wp_template',
			template.id
		);
```

Reply:
Using the built-in deleteEntityRecord should take care of invalidating the cache and doing a rerequest.

Reply:
Nice catch @david-szabo97, I didn't think of checking the sidebar. ❇️